### PR TITLE
feat: Support "Name" for AWS::Serverless::HttpApi

### PR DESCRIPTION
**Pull request by aahung**
*Issue #, if available:*

https://github.com/aws/serverless-application-model/issues/1617

*Description of changes:*

Allow customer to specify `Name` for `AWS::Serverless::HttpApi, like the rest of serverless resource types:
- AWS::Serverless::Api
- AWS::Serverless::Function
- AWS::Serverless::LayerVersion
- AWS::Serverless::SimpleTable
- AWS::Serverless::StateMachine

Why it is implemented like that?

A: `AWS::ApiGatewayV2::Api` cannot have `Body` and `Name` at the same time. As SAM-T will add `Body` to the `AWS::ApiGatewayV2::Api`, we can only set Name by updating the openapi definition (info -> title). Right now the `info -> title` is by default `!Ref: AWS::StackName`. 

Proposed documentation change:

>  Name
    A name of the API Gateway V2 Api resource.
    Note: The default name is the name of the stack. This property requires AWS SAM to modify the HttpApi resource's OpenAPI definition, to set the name field. The following two scenarios result in an error: 1) The DefinitionBody property is specified with the name field set in the OpenAPI definition (since this is a conflict that AWS SAM won't resolve), or 2) The DefinitionUri property is specified (since AWS SAM won't modify an OpenAPI definition that it retrieves from Amazon S3).
    Type: String
    Required: No
    AWS CloudFormation compatibility: This property is unique to AWS SAM and doesn't have an AWS CloudFormation equivalent.
    https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-httpapi.html

*Description of how you validated changes:*

I took the [translated template](https://github.com/aahung/serverless-application-model/blob/7f4d61671184a8f997b3d18fbdeb1daf9cf3e76b/tests/translator/output/explicit_http_api_with_name.json#L1) and deployed to my account (with stack name `http-api-test-3`):

```
  MyApi:
    Type: AWS::Serverless::HttpApi
    Properties:
      Name: MyHttpApi

  MyApiWithIntrinsicName:
    Type: AWS::Serverless::HttpApi
    Properties:
      Name: !Sub "${HttpApiFunction}-HttpApi"
```

image

*Checklist:*

- [ ] Add/update [unit tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions) using:
    - [x] Correct values
    - [x] Bad/wrong values (None, empty, wrong type, length, etc.)
    - [x] [Intrinsic Functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html)
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)
- [x] `make pr` passes
- [ ] Update documentation
- [ ] Verify transformed template deploys and application functions as expected
- [ ] Do these changes include any template validations?
    - [ ] Did the newly validated properties support intrinsics prior to adding the validations? (If unsure, please review [Intrinsic Functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html) before proceeding).
        - [ ] Does the pull request ensure that intrinsics remain functional with the new validations?

*Examples?*

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by GavinZZ — APPROVED** [review]

File: tests/translator/output/aws-cn/explicit_http_api_with_name.json [hunk 1]
```diff
@@ -87,109 +86,110 @@
...
     },
     "MyApi": {
-      "Type": "AWS::ApiGatewayV2::Api",
       "Properties": {
         "Body": {
           "info": {
-            "version": "1.0",
-            "title": "MyHttpApi"
+            "title": "MyHttpApi",
+            "version": "1.0"
```
Comment (hoffa) [inline tests/translator/output/aws-cn/explicit_http_api_with_name.json:103]: Why is this changing? I thought our formatting was deterministic?

Comment (aahung) [reply tests/translator/output/aws-cn/explicit_http_api_with_name.json:103]: This is a new file, what is changed?

Comment (hoffa) [reply tests/translator/output/aws-cn/explicit_http_api_with_name.json:103]: Commented before you pushed the new changes, ignore.

File: samtranslator/model/apigatewayv2.py [hunk 2]
```diff
@@ -17,6 +17,7 @@ class ApiGatewayV2HttpApi(Resource):
         "BodyS3Location": PropertyType(False, is_type(dict)),
         "Description": PropertyType(False, is_str()),
         "FailOnWarnings": PropertyType(False, is_type(bool)),
+        "Name": PropertyType(False, is_str()),
```
Comment (hoffa) [inline samtranslator/model/apigatewayv2.py:20]: We should use `Property` and make this any type since it's pass-through.

Comment (aahung) [reply samtranslator/model/apigatewayv2.py]: changed

**Review by hoffa — APPROVED** [review]

**Pull request merged**